### PR TITLE
Safeloader: improve the checks and contrib scripts for python unittests

### PR DESCRIPTION
**Pull request by clebergnu**
This fixes the contrib script and adds a new one that finds Python unittests.  It's also a prep work for #4625.

### CODE REVIEW

File: contrib/scripts/find-python-unittest [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import os
+import sys
+import unittest
+
+
+def get_tests(suite):
+    result = []
+    if hasattr(suite, '__iter__'):
+        for s in suite:
```
Comment (beraldoleal) [inline contrib/scripts/find-python-unittest:33]: what is s here? `for suite in suite`? `for suite in suites` ? Could you please expand this?

File: contrib/scripts/avocado-safeloader-find-python-unittest [hunk 2]
```diff
@@ -42,4 +42,8 @@ if __name__ == '__main__':
             result += ["%s.%s.%s" % (test_module_name, klass, method_name)
                        for (method_name, _, _) in methods]
     if result:
+        if os.environ.get('AVOCADO_SAFELOADER_UNITTEST_ORDER_COMPAT', False):
+            from functools import cmp_to_key
+            from unittest.util import three_way_cmp
```
Comment (beraldoleal) [inline contrib/scripts/avocado-safeloader-find-python-unittest:47]: Following the same logic on others imports, we could move this to the begin of the script.

File: .cirrus.yml [hunk 3]
```diff
@@ -71,3 +71,18 @@ fedora_selftests_task:
         matrix:
           - image: quay.io/avocado-framework/avocado-ci-fedora-32
           - image: quay.io/avocado-framework/avocado-ci-fedora-33
+
+#
+# This is a *partial* check for the behavior of the safeloader, that
+# compares it to Python's unittest behavior.  This is intended to help
+# in the resolution of https://github.com/avocado-framework/avocado/issues/4625
+# avoiding new regressions.  It should also be expanded (and/or converted
+# into a selftest) to include avocado-instrumented tests too.
```
Comment (beraldoleal) [inline .cirrus.yml:80]: Any reason for not creating this inside selftests/*, so a make check could also catch this?

**Review by beraldoleal — COMMENTED** [review]
Hi @clebergnu , thank you for this.

I left a few comments for you.

File: .cirrus.yml (hunk 3 above)
Comment (clebergnu) [reply .cirrus.yml:80]: No, I even mentioned that.  Doing that in v2.

File: contrib/scripts/find-python-unittest (hunk 1 above)
Comment (clebergnu) [reply contrib/scripts/find-python-unittest:33]: The suite can either contain other suites, or tests.  I'm renaming this to `suite_or_test` in v2.

**Comment by clebergnu** [comment]
Closing in favor of #4638

**Pull request closed without merging**